python/ray/data/context.py

```diff
@@ -11,7 +11,7 @@
 _context_lock = threading.Lock()
 
 # The max target block size in bytes for reads and transformations.
-DEFAULT_TARGET_MAX_BLOCK_SIZE = 2048 * 1024 * 1024
+DEFAULT_TARGET_MAX_BLOCK_SIZE = 512 * 1024 * 1024
```

Comment:
Is it possible to make this user-configurable per dataset?

Reply:
Yes, you can configure it via the DatasetContext.

Reply:
Ah I see. It would be nice if the user could pass it as a top-level arg like `parallelism`, but maybe we can do this as a follow-up?

Reply:
I see, agree that could be nice. I think we should do this when enabling block splitting by default (and maybe deprecating parallelism at the same time. That way, the story becomes pretty simple: parallelism is auto-detected usually, and the block size can be reduced if needed if you're running into OOMs.